presto-main/src/main/java/com/facebook/presto/operator/WindowOperator.java

```diff
@@ -924,17 +924,19 @@ static int findEndPosition(int startPosition, int endPosition, BiPredicate<Integ
         checkArgument(startPosition < endPosition, "startPosition (%s) must be less than endPosition (%s)", startPosition, endPosition);
 
         int left = startPosition;
-        int right = endPosition;
-
-        while (left + 1 < right) {
-            int middle = (left + right) >>> 1;
-
-            if (comparator.test(startPosition, middle)) {
-                left = middle;
-            }
-            else {
-                right = middle;
+        int right = left + 1;
+
+        while (right < endPosition) {
+            int distance = 1;
```

Comment:
Nit : Remove declaration of int distance outside of the loop. Just doing a reassignment here is preferred.

Reply:
Well, I don't see the necessity of this. Actually I think it is better to declare it inside the loop rather than outside.

From semantic point, the `distance` variable is reassigned at the beginning of the each loop and is only used inside of the loop, so the liveness **should** be narrowed to the loop scope because it acts as a local variable of the loop rather than the whole function.
From performance point, the location of `distance` variable declaration wouldn't change the liveness analysis of the compiler (the liveness of `distance` is always from assignment of the beginning of loop to the last usage of the end of loop), and wouldn't change the stack consumption.